media/editor/dataDisplay.tsx

```diff
@@ -571,24 +662,49 @@ const DataCell: React.FC<{
 				return;
 			}
 
-			if (isChar) {
-				// b is final
-			} else if (firstOctetOfEdit !== undefined) {
-				newValue = (firstOctetOfEdit << 4) | newValue;
-			} else {
-				return setFirstOctetOfEdit(newValue);
-			}
+			if (editMode === HexDocumentEditOp.Insert) {
+				if (isChar) {
+					ctx.focusedElement = ctx.focusedElement?.shift(1);
+					// Finishes byte insertion
+				} else if (firstOctetOfEdit !== undefined) {
+					ctx.edit({
```

Comment:
In insert mode, shouldn't this generate an insert operation rather than a replace operation?